### PR TITLE
fix: Improve Pipeline workflow unit test branch coverage

### PR DESCRIPTION
**Pull request by qidewenwhen**
*Description of changes:* 
- Fix tiny bugs among multiple classes
- Add more unit tests to improve branch coverage
- Exclude script files and deprecated classes from the coverage calculation

*Testing done:* Unit tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/workflow/_repack_model.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 from distutils.dir_util import copy_tree
 
 
-def repack(inference_script, model_archive, dependencies=None, source_dir=None):
+def repack(inference_script, model_archive, dependencies=None, source_dir=None):  # pragma: no cover
```
Comment (jerrypeng7773) [inline src/sagemaker/workflow/_repack_model.py:37]: why we want to mark this function as "no cover"?

File: src/sagemaker/workflow/condition_step.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def properties(self):
 
 
 @attr.s
-class JsonGet(Expression):
+class JsonGet(Expression):  # pragma: no cover
```
Comment (jerrypeng7773) [inline src/sagemaker/workflow/condition_step.py:98]: ditto

Comment (jerrypeng7773) [inline src/sagemaker/workflow/condition_step.py:98]: this make senses, as we deprecated the JsonGet defined in `condition_step.py`

File: src/sagemaker/workflow/_repack_model.py (hunk 1 above)
Comment (qidewenwhen) [reply src/sagemaker/workflow/_repack_model.py:37]: This function is internally used and only used in a script to repack model. It does have corresponding [unit test file](https://github.com/aws/sagemaker-python-sdk/blob/dev/tests/unit/sagemaker/workflow/test_repack_model_script.py) but all of them are skipped due to 
```
This test operates on the root file system and will likely fail due to permission errors.
```

Therefore, I guess we don't need this calculated in the coverage report

File: src/sagemaker/workflow/condition_step.py (hunk 2 above)
Comment (qidewenwhen) [reply src/sagemaker/workflow/condition_step.py:98]: Yes. This PR also removed the usage of such JsonGet defined in `condition_step.py`

File: tests/unit/sagemaker/workflow/test_functions.py [hunk 3]
```diff
@@ -97,3 +99,23 @@ def test_json_get_expressions():
             "Path": "my-json-path",
         },
     }
+
+
+def test_json_get_expressions_without_invalid_step_name():
```
Comment (staubhp) [inline tests/unit/sagemaker/workflow/test_functions.py:104]: nit `test_json_get_expressions_with_invalid_step_name`

Comment (qidewenwhen) [reply tests/unit/sagemaker/workflow/test_functions.py]: Good catch! Fixed in new commit.

**Review by jerrypeng7773 — APPROVED** [review]

**Review by staubhp — APPROVED** [review]

File: src/sagemaker/workflow/step_collections.py [hunk 4]
```diff
@@ -318,15 +318,15 @@ def __init__(
         """
         steps = []
         if "entry_point" in kwargs:
-            entry_point = kwargs["entry_point"]
-            source_dir = kwargs.get("source_dir")
-            dependencies = kwargs.get("dependencies")
+            entry_point = kwargs.get("entry_point")
```
Comment (navinsoni) [inline src/sagemaker/workflow/step_collections.py:321]: do we have any default value here if "entry_point" is not present here

Comment (qidewenwhen) [reply src/sagemaker/workflow/step_collections.py:321]: Thanks for reviewing! 
As in previous line (line 320), we've already checked whether the `entry_point` is in `kwargs`, then if it goes into this `if` branch, we can be sure that the "entry_point" should exist.
```
if "entry_point" in kwargs:
  entry_point = kwargs.get("entry_point")
  ...
```

File: tests/integ/test_workflow.py [hunk 5]
```diff
@@ -67,7 +67,6 @@
     ConditionLessThanOrEqualTo,
 )
 from sagemaker.workflow.condition_step import ConditionStep
-from sagemaker.workflow.condition_step import JsonGet as ConditionStepJsonGet
```
Comment (qidewenwhen) [inline tests/integ/test_workflow.py]: The JsonGet defined in `condition_step.py` is deprecated so update to the latest one

File: src/sagemaker/workflow/step_collections.py (hunk 4 above)
Comment (qidewenwhen) [reply src/sagemaker/workflow/step_collections.py]: Added default value in the new commit

File: src/sagemaker/workflow/step_collections.py [hunk 6]
```diff
@@ -323,15 +323,15 @@ def __init__(
         """
         steps = []
         if "entry_point" in kwargs:
-            entry_point = kwargs["entry_point"]
-            source_dir = kwargs.get("source_dir")
-            dependencies = kwargs.get("dependencies")
+            entry_point = kwargs.get("entry_point", None)
```
Comment (BasilBeirouti) [inline src/sagemaker/workflow/step_collections.py:326]: much neater

**Review by BasilBeirouti — APPROVED** [review]

File: src/sagemaker/workflow/step_collections.py (hunk 6 above)
Comment (navinsoni) [reply src/sagemaker/workflow/step_collections.py:326]: +1

**Review by navinsoni — APPROVED** [review]

**Pull request merged**